test/server/cards/02_SHD/units/EmboStoicAndResolute.spec.ts

```diff
@@ -0,0 +1,69 @@
+        describe('Embo\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['embo#stoic-and-resolute', { card: 'echo-base-defender', damage: 2 }],
+                        spaceArena: ['green-squadron-awing']
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine', 'consular-security-force', 'swoop-racer', 'superlaser-technician'],
+                    }
+                });
+            });
+
+            it('should heal 2 damage from a unit when he kill someone and survives', function () {
+                const { context } = contextRef;
+
+                function reset(resetDamage = true) {
+                    context.embo.exhausted = false;
+                    if (resetDamage) {
+                        context.embo.damage = 0;
+                    }
+                    context.player2.passAction();
+                }
+
+                // kill battlefield marine, should heal 2 from a unit
+                context.player1.clickCard(context.embo);
+                context.player1.clickCard(context.battlefieldMarine);
+                expect(context.player1).toBeAbleToSelectExactly([context.embo, context.echoBaseDefender, context.consularSecurityForce, context.swoopRacer, context.greenSquadronAwing, context.superlaserTechnician]);
+                expect(context.player1).toHaveChooseNoTargetButton();
+                context.player1.setDistributeHealingPromptState(new Map([
+                    [context.echoBaseDefender, 2],
+                ]));
+
+                // heal echo base defender
+                expect(context.echoBaseDefender.damage).toBe(0);
+                expect(context.embo.damage).toBe(3);
+
+                reset();
+
+                // defender survives, nothing happen
+                context.player1.clickCard(context.embo);
+                context.player1.clickCard(context.consularSecurityForce);
+                expect(context.player2).toBeActivePlayer();
+
+                reset();
+
+                // defender is defeated but not in discard
+                context.player1.clickCard(context.embo);
+                context.player1.clickCard(context.superlaserTechnician);
+                context.player2.clickPrompt('Put Superlaser Technician into play as a resource and ready it');
+                expect(context.player1).toBeAbleToSelectExactly([context.embo, context.echoBaseDefender, context.consularSecurityForce, context.swoopRacer, context.greenSquadronAwing]);
+                expect(context.player1).toHaveChooseNoTargetButton();
+                context.player1.setDistributeHealingPromptState(new Map([
+                    [context.echoBaseDefender, 2],
+                ]));
+
+                reset(false);
+
+                // embo does not complete attack, nothing happen
```

Comment:
Couple more cases it would be good to have:

- Enemy unit attacks Embo and is defeated, ability doesn't trigger
- Defender is defeated before combat damage due to some effect like Vambrace Flamethrower, Embo ability still triggers (this can be a separate setupTest potentially since it involves an upgrade)

Also re: the comment on Embo, we should check the following more complicated case (could also be its own setupTest). These are the steps:

1. Enemy unit is defeated by Vanquish or something
2. Opponent uses Emperor's Legion to return it to hand
3. Opponent plays the unit
4. Embo attacks and does not defeat the unit
5. Confirm that his ability does not trigger since it was not defeated during this attack (even though it was defeated earlier in the phase)